pilot/pkg/xds/workloadentry.go

```diff
@@ -101,15 +107,11 @@ func (sg *InternalGen) QueueUnregisterWorkload(proxy *model.Proxy) {
 	}
 
 	// unset controller, set disconnect time
-	cfg := sg.Store.Get(gvk.WorkloadEntry, entryName, proxy.Metadata.Namespace)
-	if cfg == nil {
-		// we failed to create the workload entry in the first place
-		return
-	}
-	wle := cfg.DeepCopy()
-	delete(wle.Annotations, WorkloadControllerAnnotation)
-	wle.Annotations[DisconnectedAtAnnotation] = strconv.FormatInt(time.Now().UnixNano(), 10)
-	_, err := sg.Store.Update(wle)
+	_, err := sg.Store.Patch(gvk.WorkloadEntry, entryName, proxy.Metadata.Namespace, func(cfg config.Config) config.Config {
+		delete(cfg.Annotations, WorkloadControllerAnnotation)
```

Comment:
I think patch here is intended so if another Istiod' has already taken ownership we do not overwrite.

Reply:
Right about the "so we don't overwrite", but we want to use "Update" since that fails on conflict.